Stdlib/Data/List.juvix

```diff
@@ -137,7 +146,82 @@ transpose xss := map head xss :: transpose (map tail xss);
 any : {A : Type} → (A → Bool) → List A → Bool;
 any f xs := foldl or false (map f xs);
 
+--- 𝒪(𝓃). Returns ;true; if all elements of the ;List; satisfy the predicate.
+all : {A : Type} -> (A -> Bool) -> List A -> Bool;
+all f xs := foldl and true (map f xs);
+
 --- 𝒪(1). Returns ;true; if the ;List; is empty.
 null : {A : Type} → List A → Bool;
 null nil := true;
 null _ := false;
+
+--- 𝒪(min(𝓂, 𝓃)). Returns a list containing the results of applying a function
+--- to each pair of elements from the input lists.
+zipWith :
+  {A : Type}
+    -> {B : Type}
+    -> {C : Type}
+    -> (A -> B -> C)
+    -> List A
+    -> List B
+    -> List C;
+zipWith {A} {B} {C} f as bs :=
+  let
+    go : List A -> List B -> List C -> List C;
+    go _ nil acc := acc;
+    go nil _ acc := acc;
+    go (a :: az) (b :: bz) acc := go az bz (snoc acc (f a b));
+  in go as bs nil;
+
+--- 𝒪(min(𝓂, 𝓃)). Returns a list of pairs formed from the input lists.
+zip :
+  {A : Type}
+    -> {B : Type}
+    -> List A
+    -> List B
+    -> List (A × B);
+zip {A} {B} as bs :=
+  let
+    go : List A -> List B -> List (A × B) -> List (A × B);
+    go _ nil acc := acc;
+    go nil _ acc := acc;
+    go (a :: az) (b :: bz) acc := go az bz (snoc acc (a, b));
+  in go as bs nil;
+
+--- 𝒪(𝓃 log 𝓃). Sorts a list of elements in ascending order using the MergeSort
+--- algorithm.
+mergeSort :
+  {A : Type} → (A → A → Ordering) → List A → List A;
+mergeSort {A} cmp :=
+  let
+    terminating
+    go : List A -> List A;
+    go nil := nil;
+    go xs@(_ :: nil) := xs;
+    go xs :=
+      let
+        splitXs :
+            List A × List A := splitAt (div (length xs) 2) xs;
+        left : List A := fst splitXs;
+        right : List A := snd splitXs;
+      in merge cmp (go left) (go right);
+  in go;
+
+--- 𝒪(𝓃²). Sorts a list of elements in ascending order using te QuickSort
```

Comment:
Maybe we should add that expected cost is 𝒪(𝓃 log 𝓃)

Reply:
Good point, I've added this to the judoc